Petr Dolejš - cleancode - NNPTP21-1

DOL7JS opened the pull request:
Renamed branch to dev

DOL7JS committed:
Petr Dolejš, st58223
Moved class ComplexNumber, Polynom.
Created and refactored class NewtonFractal.

DOL7JS committed:
Edited ParseArgument() - take care of situation when there are no input parameters

DOL7JS committed:
Renaming branch to dev

rodi0878 commented on INPTPZ1/NewtonFractal.cs:
```diff
@@ -0,0 +1,163 @@
+        private float FindEquationSolutionWithNewtonsIteration(ref ComplexNumber complexNumber)
+        {
+            float iteration = 0;
+            for (int q = 0; q < 30; q++)
```
why q?

rodi0878 commented on INPTPZ1/NewtonFractal.cs:
```diff
@@ -0,0 +1,163 @@
+        }
+        private float FindEquationSolutionWithNewtonsIteration(ref ComplexNumber complexNumber)
+        {
+            float iteration = 0;
```
why float? i see only incrementation and not a single float/double operation

rodi0878 commented on INPTPZ1/NewtonFractal.cs:
```diff
@@ -0,0 +1,163 @@
+        private int FindRootNumberSolution(ComplexNumber complexNumber)
+        {
+            var known = false;
+            var id = 0;
```
index not id

rodi0878 commented on INPTPZ1/NewtonFractal.cs:
```diff
@@ -0,0 +1,163 @@
+        private void ColorizeOnePixel(int i, int j, float it, int id)
+        {
+            var colorForPixel = COLORS[id % COLORS.Length];
+            colorForPixel = Color.FromArgb(colorForPixel.R, colorForPixel.G, colorForPixel.B);
```
this line does nothing - remove

rodi0878 commented on INPTPZ1/Program.cs:
```diff
@@ -25,309 +13,10 @@ class Program
-
-                    bmp.Save(output ?? "../../../out.png");
-            //Console.ReadKey();
+            NewtonFractal newtonFractal = new NewtonFractal(args);
```
i really dont like hiding all the work in the constructor. its not a good style; please create atleast one method which will do real work and call it. constructor is only for setting up of object and not for these calculations

rodi0878 reviewed (changes requested):
There are few things that still need cleaning up. Please look at them and update your patch.

DOL7JS committed:
Renamed variable in for cycle - q => i
Refactored float to int
Renamed id to index
Removed useless line
Created new method

DOL7JS commented:
Refactored

- Renamed variable in for cycle - q => i
- Refactored float to int
- Renamed id to index
- Removed useless line
- Created new method

rodi0878 commented:
accepted
